### ISSUE TITLE
🇩🇪 German Language Support — Community Tracking Issue

### ISSUE THREAD

**Issue by leonbubova** [moonshine-ai/moonshine#141]
This issue is a gathering place for everyone interested in bringing German language support to Moonshine — researchers, developers, and data contributors alike.

German is underserved in the edge STT space. Moonshine's architecture is a great fit, but there's no German model yet.

Comment if you're:
working on a German fine-tune
know of useful German audio datasets
interested in following progress

Let's coordinate here instead of scattered issues and PRs.

**Comment by fosple** [comment]
I trained a first model and got the WER for German down to 13.16% (Fleurs DE) / 11.95 % (MCV DE) for moonshine-streaming-small - not yet where I would like it to be.

To get it down even further I would need access to more GPU compute resources. If you have access to some compute cluster (e.g. of H100 GPUs) feel free to reach out :)

**Comment by dattazigzag** [comment]
> I trained a first model and got the WER for German down to 13.16% (Fleurs DE) / 11.95 % (MCV DE) for moonshine-streaming-small - not yet where I would like it to be.
> 
> To get it down even further I would need access to more GPU compute resources. If you have access to some compute cluster (e.g. of H100 GPUs) feel free to reach out :)

Also, few quick questions (popped in my head during lunch):

1. Which dataset did you use to train on? 
2. Can we try your model that you have currently (maybe if it is uploaded to hf) as is?

**Comment by leonbubova** [comment]
@SimonSchroffner or something like https://lambda.ai/ but yes i would be down for that as well!

I've uploaded my findings and research in the meantime as well here: https://github.com/leonbubova/moonshine-german

I haven't reached more than Proof of concept stage yet though. WER ~50%

![Screenshot_20260410-003237.png](https://github.com/user-attachments/assets/60f37353-e64d-443e-bc21-b73334e91446)

**Comment by malkozajche** [comment]
Patiently waiting for German support, fingers crossed. 🤞

**Comment by AlbrinMemedi** [comment]
I am also interested in this.  I am an Embeeded Software Developer and i am willing to help if possible.

**Comment by pierre-cheneau** [comment]
> Maybe something as inspiration:
> 
> https://github.com/pierre-cheneau/finetune-moonshine-asr

Hey! The creator of this repo here, if I can be of any help, please feel free to ask!

**Comment by akramer-zibra** [comment]
Any news on this @fosple? Would love to use your work

**Comment by GeorgeS2019** [comment]
@leonbubova 
@pierre-cheneau
```
https://github.com/leonbubova/moonshine-german
```

sorry I just found out it is [WIP](https://github.com/leonbubova/moonshine-german)

**Comment by ChrisMicro** [comment]
Probably it is best to start with the voice synthesizer. There is a github repo [g2p_de](https://github.com/gooofy/g2p_de) for the German Grapheme To Phoneme Conversion. I hope this fits in the current implementation of the English synth.

**Comment by leonbubova** [comment]
Status update from my side, plus two questions.

**Where I am:** I rebuilt my training pipeline after finding that most of my earlier runs never produced a model — they either died during preprocessing or crashed before the first training step (a YAML parsing issue silently turned the learning rate into a string, so the LR scheduler failed immediately). My verified baseline is 54.8% WER on MLS German with `moonshine-tiny` (15h, 200 steps) — well behind what others here have reported.

**@fosple** — your 13.16% FLEURS DE / 11.95% MCV DE on `moonshine-streaming-small` is the best German result in this thread. If you're willing to share: which datasets and roughly how many hours; the learning rate, schedule and number of steps; and the effective batch size you trained at. Also whether you evaluated on the full FLEURS/MCV test splits, so I can score the same way instead of against my own held-out split.

**@pierre-cheneau** — thanks for the offer, and for the toolkit, which I'm training with. Your MLS French config uses lr 3.0e-4 with `constant_with_warmup`, 8000 steps at effective batch 64 on `moonshine-tiny`. I had been running 5e-5, which I now believe is my main problem. Does 3e-4 hold up on a smaller corpus (60–100h rather than your ~60k samples), or would you scale it down? And how much did curriculum learning and `group_by_length` actually buy you?

**What I can contribute back:** my pipeline runs on a single free-tier Kaggle T4, so it's reproducible without a GPU cluster. I've also verified the Android deployment path for non-streaming Moonshine: HF checkpoint → ONNX → `.ort` produces exactly the three files the `moonshine-voice` SDK loads (`encoder_model.ort`, `decoder_model_merged.ort`, `tokenizer.bin`), and `tokenizer.bin` is byte-identical across the official bundles, so a fine-tune doesn't need a new one. Happy to write that up if it's useful.

**Comment by GeorgeS2019** [comment]
@pierre-cheneau

Can you try **moonshine-tiny-de**

**Comment by petewarden** [comment]
I've [just launched our first in-house German models, in Small and Tiny streaming variants](https://moonshine-voice.readthedocs.io/en/stable/models/available-models/). These are still being actively developed and I'd very much like to reduce the WER and bring out a Medium version, so any feedback from native speakers would be very much appreciated.

**Comment by GeorgeS2019** [comment]
@petewarden 

Could you consider sharing ONNX version?


| Language | Model           | Parameters   | WER   | License |
|----------|------------------|--------------|-------|---------|
| German   | Small Streaming  | 123 million  | 7.5%  | MIT     |
| German   | Tiny Streaming   | 34 million   | 12.0% | MIT     |

**Comment by petewarden** [comment]
We don't ship any .onnx files, we treat those as intermediates, but you can find [a conversion recipe in the domain customization docs](https://moonshine-voice.readthedocs.io/en/stable/models/domain-customization/#shipping-the-result). You'll need to do something like this with a downloaded .safetensors and tokenizer:

```bash
python -m moonshine_voice.lora --export \
    --model moonshine-ai/moonshine-streaming-tiny-de \
    --output-dir ./float \
    --tokenizer-bin /path/to/published/tokenizer.bin 
```

**Comment by GeorgeS2019** [comment]
@petewarden 
moonshine-ai/moonshine-streaming-tiny-de not shipped with tokenizer.bin

/path/to/published/tokenizer.bin

**Comment by GeorgeS2019** [comment]
### 1. Moonshine Streaming Tiny — German
[moonshine-ai/moonshine-streaming-tiny-de ](https://huggingface.co/moonshine-ai/moonshine-streaming-tiny-de?utm_source=copilot.com)


| Language | Model           | Parameters   | WER   | License |
|----------|------------------|--------------|-------|---------|
| German   | Small Streaming  | 123 million  | 7.5%  | MIT     |
| German   | Tiny Streaming   | 34 million   | 12.0% | MIT     |


### 2. Moonshine Tiny — German (fine‑tuned) 
- [dattazigzag/moonshine-tiny-de](https://huggingface.co/dattazigzag/moonshine-tiny-de)
Base model: UsefulSensors/moonshine-tiny

Params: 27M
Training data: MLS German (~1,967 hours)
WER: 36.7%
Notes: Read‑speech only (audiobooks), not conversational
License: MIT

### 3. [German Moonshine Base](https://huggingface.co/fidoriel/moonshine-base-de) (fine‑tuned)
Moonshine Base — German (fidoriel)
Repo: fidoriel/moonshine-base-de  
(Referenced via GGUF conversion below)

Params: 61.5M

WER: 6.9% (Common Voice 22 German)

Notes: High‑quality German finetune of Moonshine Base

### 4.. [GGUF German Moonshine Models ](https://huggingface.co/cstr/moonshine-base-de-fidoriel-GGUF?utm_source=copilot.com)(CPU‑optimized)Moonshine Base — German (GGUF)
Repo: cstr/moonshine-base-de-fidoriel-GGUF 

Type: GGUF quantized versions
Params: 61.5M
Variants:
moonshine-base-de-fidoriel-f16.gguf (118 MB)
moonshine-base-de-fidoriel-q4_k.gguf (39 MB)
WER: 6.9%

Notes: Best German Moonshine quality available; works with CrispASR

--- 
| Model | Repo | Size | Type | WER | Notes |
|-------|-------|-------|-------|-------|-------|
| moonshine-ai/moonshine-streaming-small-de | 123M | Streaming ASR | 7.5% | Official German streaming model |
| moonshine-ai/moonshine-streaming-tiny-de | 34M | Streaming ASR | 12.0% | Official German streaming model |
| dattazigzag/moonshine-tiny-de | 27M | ASR | 36.7% | MLS audiobook German |
| fidoriel/moonshine-base-de | 61.5M | ASR | 6.9% | High‑quality German finetune |
| cstr/moonshine-base-de-fidoriel-GGUF | 61.5M | GGUF | 6.9% | CPU‑optimized, CrispASR compatible |

**Comment by petewarden** [comment]
Thanks, this is very helpful, and was a good prompt for me to look at the overall context of available German models. I scored checkpoints listed above on the same panel we use for our own training, so the numbers are comparable.

**Panel (`suite_de`).** Full test splits, not a sample:

- FLEURS German (`google/fleurs`, `de_de` / `test`) — 862 utterances, short prompted read speech
- Multilingual LibriSpeech German (`facebook/multilingual_librispeech`, `german` / `test`) — 3,394 utterances, LibriVox audiobooks

**Headline** is the unweighted macro of those two WERs. **Normalizer** is a German `BasicTextNormalizer` that keeps umlauts and ß (the English Whisper normalizer would strip them and inflate WER). **Decoding** is greedy, float32; we did not use beam search, even where a repo ships `num_beams=5`.

| Model | Architecture | FLEURS | MLS | **Macro** | Previously reported |
|---|---|---:|---:|---:|---|
| [`moonshine-ai/moonshine-streaming-tiny-de`](https://huggingface.co/moonshine-ai/moonshine-streaming-tiny-de) | Streaming Tiny (27M) | **11.60** | **10.44** | **11.02** | 12.0% in the model docs (that figure is the quantized CDN build) |
| [`moonshine-ai/moonshine-streaming-small-de`](https://huggingface.co/moonshine-ai/moonshine-streaming-small-de) | Streaming Small (113M) | ~7.5 | ~8.1 | **7.63** | 7.5% in the model docs |
| [`fidoriel/moonshine-tiny-de`](https://huggingface.co/fidoriel/moonshine-tiny-de) | Original Tiny (27M) | 16.83 | 18.59 | 17.71 | 11.4% on Common Voice 22 |
| [`fidoriel/moonshine-base-de`](https://huggingface.co/fidoriel/moonshine-base-de) | Original Base (61.5M) | 13.08 | 25.89 | 19.49 | 6.9% on Common Voice 22 |
| [`dattazigzag/moonshine-tiny-de`](https://huggingface.co/dattazigzag/moonshine-tiny-de) | Original Tiny (27M) | 44.00 | 22.03 | 33.01 | 36.7% on MLS German |

The GGUF/ONNX repos are quantized copies of the fidoriel weights, so they are not a separate result. I could not score fosple’s streaming-small run (13.16% FLEURS / 11.95% MCV) because I couldn't find those weights.

A couple of notes on the previously reported figures, which are real measurements but not the same test:

- **Common Voice 22** (fidoriel) is short crowdsourced read speech. It is much closer to FLEURS than to MLS, which is why Base looks strong there (6.9%) and then jumps to 25.9% on audiobooks.
- **MLS train** (dattazigzag) is the audiobook corpus; their MLS number is in-domain, and FLEURS (which that run never saw) is 44%. Our greedy + German-normalizer MLS reading for that checkpoint is 22.0%, against the 36.7% on the card — likely a mix of beam/normalizer/length-cap differences rather than a different model.
- **MCV / FLEURS** as reported by fosple is a different pair again (and a Small, not a Tiny).

I definitely welcome community models, and I'm working on open sourcing more of our training infrastructure which will hopefully make creating them easier long-term. The current community models listed above report WERs based on the domain they've been trained on, but tend to underperform on out-of-domain test sets. The official models use a mix of domains since they're intended to be widely applicable. 

I'm still not happy with our results yet, especially compared to Whisper Large V3 on German, and I'm actively trying to reduce our WERs. 

Please let me know if I'm missing something, leider spreche ich nur ein wenig Deutsch, so native-speaker feedback on the streaming Tiny/Small checkpoints is very welcome!

**Comment by GeorgeS2019** [comment]
@petewarden 
I need csharp support and so far the only success using moonshine German model  is through [CrispASR](https://github.com/CrispStrobe/CrispASR/issues/291#issuecomment-5450951006)
